### PR TITLE
prov/verbs CM cleanups

### PR DESCRIPTION
**Pull request by fzago-cray**
No bug fix here. Just some cleanups.

### CODE REVIEW

File: prov/verbs/src/ep_rdm/verbs_rdm_cm.c [hunk 1]
```diff
@@ -815,72 +817,51 @@ fi_ibv_rdm_process_event(struct rdma_cm_event *event, struct fi_ibv_rdm_ep *ep)
...
-				pthread_mutex_unlock(&ep->domain->rdm_cm->cm_lock);
-				ret = -FI_ENOMEM;
-				break;
-			}
-			memcpy(data, event->param.conn.private_data,
-				      event->param.conn.private_data_len);
-			event_copy.param.conn.private_data = data;
-			event_copy.param.conn.private_data_len =
-			    event->param.conn.private_data_len;
+			size_t len = MIN(event->param.conn.private_data_len,
```
Comment (gladkovdmitry17) [inline prov/verbs/src/ep_rdm/verbs_rdm_cm.c]: Seems, it's impossible case in a normal situation. The private data is always `sizeof(struct fi_conn_priv_params)`, isn't it?
But to be safe here, I'm okay with it

Comment (fzago-cray) [reply prov/verbs/src/ep_rdm/verbs_rdm_cm.c]: One would expect, but it's not what I saw. fi_conn_priv_params is 40 bytes, but every length I got was between 56 and 190 bytes. No idea why. Hence the MIN.

Comment (gladkovdmitry17) [reply prov/verbs/src/ep_rdm/verbs_rdm_cm.c]: > No idea why

Thanks for the explanation. Seems something is hidden by rdma cm library :)

Comment (shefty) [reply prov/verbs/src/ep_rdm/verbs_rdm_cm.c]: The underlying CMs always carry the full amount of private data available in the CM protocols.  For IB, this varies from 56 to 190 bytes depending on which message was sent.  There's no way to convey how much of what is received was initialized by the sender.

**Pull request merged**